### PR TITLE
issue 727 boundary changes quick fix

### LINKED ISSUE #727: Integrate new source of constituencies
Blocked by #726 

Breaks into two parts

## Adapt existing database/code to lack of ONS_ID

Means code adjustments in the following areas

- polls (predictions or past results, currently electoral calculus) are indexed partially by constituency_ons_id
- recommendations (TV) are indexed partially by constituency_ons_id
- users own constituency noted by by constituency_ons_id
- the API (for sites linking directly to swapmyvote) prepopulates the form based on ONS_ID

### PR DESCRIPTION
**Pull request by baob**
Fixes #727

This should allow the site to go public, but no constituency guidance (either previous/predicted polls, or tactical vote recommendations) as yet.

I propose we get this on the staging site ASAP for review.

- update the postcodes.io lookup to ge2024 boundaries
- new constituencies downloaded from mysociety
- do not seed the polls data, need something that will work with the new constituencies
- for now always hide polls - new constituencies means no immediate tie-in to polling data
- do not seed the recommendations data, need something that will work with the new constituencies
- fix html references to the constituency data
- fix: hardcoded date and more on poll results
- note: ELECTION_TYPE is always defaulted sensibly - except when seeding the database
- replace my society constituencies with latest version, 0.1.4

### CODE REVIEW

File: app/views/user/swaps/_swap_profile_inner.html.haml [hunk 1]
```diff
@@ -23,5 +23,5 @@
 - if !hide_polls? && !other_user.constituency.nil?
   .profile-poll
     .text-center.small
-      GE2019 results for #{other_user.constituency.try(:name) or "Unknown"}
+      Predicted GE2024 results for #{other_user.constituency.try(:name) or "Unknown"}
```
Comment (aspiers) [inline app/views/user/swaps/_swap_profile_inner.html.haml:26]: Might be worth getting this from `election_year` or similar at some point? Not important though.

**Review by aspiers — DISMISSED** [review]

**Review by aspiers — APPROVED** [review]

File: app/views/user/swaps/_swap_profile_inner.html.haml (hunk 1 above)
Comment (baob) [reply app/views/user/swaps/_swap_profile_inner.html.haml:26]: Right. I was most concerned with the change from previous results to predicted results. But yes, could be tweaked.

**Comment by baob** [comment]
> @[baob](https://github.com/baob) baob dismissed [aspiers](https://github.com/aspiers)’s [stale review](https://github.com/swapmyvote/swapmyvote/pull/747#pullrequestreview-2078958443) [2 minutes ago](https://github.com/swapmyvote/swapmyvote/pull/747#event-12932640752)
The merge-base changed after approval.

I did ? I don't remember doing that ?

**Comment by aspiers** [comment]
I think it must have been automated

**Comment by aspiers** [comment]
Maybe you need to rebase?

**Review by aspiers — APPROVED** [review]

**Pull request merged**